pkg/crc/preflight/preflight_checks_network_linux.go

```diff
@@ -248,6 +248,15 @@ func checkSystemdResolvedIsRunning() error {
 }
 
 func checkCrcNetworkManagerDispatcherFile() error {
+	// In case user upgrade from f-32 to f-33 the dnsmasq config for NM still
+	// exist and need to be removed.
+	if err := removeCrcNetworkManagerConfig(); err != nil {
+		logging.Debugf("%s: not present.", crcNetworkManagerConfigPath)
+	}
+	if err := removeCrcDnsmasqConfigFile(); err != nil {
+		logging.Debugf("%s: not present.", crcDnsmasqConfigPath)
+	}
+
```

Comment:
I'd have a slight preference for doing this in a separate preflight check to make it clearer that it's cleanup for older releases ? You can indicate explicitly that this can be removed once f34 is released as f32 will be EOL'ed then. Though the same code may be needed if RHEL ever switches from NM+dnsmasq to NM+systemd-resolved.
And last but not least, this feels like a 'cleanup' thing rather than something we want to  transparently support at setup time?

Reply:
> this feels like a 'cleanup' thing rather than something we want to transparently support at setup time?

Most of the time user run the setup, be it a new release or running on a fresh system. cleanup tends to there to remove the changes which is done by setup side, this change is quite unusual since we are not doing it for new versions but carry forwarded from old releases. I still think having this here make more sense then putting it to cleanup side.

Reply:
Also when you mean by having it only a separate preflight clean up check, you mean something like below?

```patch
diff --git a/pkg/crc/preflight/preflight_checks_network_linux.go b/pkg/crc/preflight/preflight_checks_network_linux.go
index e678b278..61e237f2 100644
--- a/pkg/crc/preflight/preflight_checks_network_linux.go
+++ b/pkg/crc/preflight/preflight_checks_network_linux.go
@@ -38,6 +38,13 @@ var nmPreflightChecks = [...]Check{
        },
 }
 
+var olderReleaseCleanupChecks = [...]Check{
+       {
+               cleanupDescription: "Removing file from older os releases",
+               cleanup:            removeOlderConfigsFromOS,
+       },
+}
+
 var dnsmasqPreflightChecks = [...]Check{
        {
                configKeySuffix:    "check-network-manager-config",
@@ -248,15 +255,6 @@ func checkSystemdResolvedIsRunning() error {
 }
 
 func checkCrcNetworkManagerDispatcherFile() error {
        logging.Debug("Checking NetworkManager dispatcher file for crc network")
        err := crcos.FileContentMatches(crcNetworkManagerDispatcherPath, []byte(crcNetworkManagerDispatcherConfig))
        if err != nil {
@@ -280,3 +278,15 @@ func fixCrcNetworkManagerDispatcherFile() error {
 func removeCrcNetworkManagerDispatcherFile() error {
        return removeNetworkManagerConfigFile(crcNetworkManagerDispatcherPath)
 }
+
+func removeOlderConfigsFromOS() error{
+       // In case user upgrade from f-32 to f-33 the dnsmasq config for NM still
+       // exist and need to be removed.
+       if err := removeCrcNetworkManagerConfig(); err != nil {
+               logging.Debugf("%s: not present.", crcNetworkManagerConfigPath)
+       }
+       if err := removeCrcDnsmasqConfigFile(); err != nil {
+       logging.Debugf("%s: not present.", crcDnsmasqConfigPath)
+       }
+       return nil
+}
\ No newline at end of file
diff --git a/pkg/crc/preflight/preflight_linux.go b/pkg/crc/preflight/preflight_linux.go
index 246465c9..6ce3f71d 100644
--- a/pkg/crc/preflight/preflight_linux.go
+++ b/pkg/crc/preflight/preflight_linux.go
@@ -180,6 +180,7 @@ func getNetworkChecksForDistro(distro *linux.OsRelease, networkMode network.Mode
                checks = append(checks, nmPreflightChecks[:]...)
                if usesSystemdResolved(distro) {
                        checks = append(checks, systemdResolvedPreflightChecks[:]...)
+                       checks = append(checks, olderReleaseCleanupChecks[:]...)
                } else {
                        checks = append(checks, dnsmasqPreflightChecks[:]...)
                }

```

Reply:
> Most of the time user run the setup, 

Setup should only be needed for the first run and after a cleanup.